fl4health/model_bases/fenda_base.py

```diff
@@ -1,58 +1,43 @@
-from abc import ABC, abstractmethod
-from enum import Enum
 from typing import Dict, List, Tuple
 
 import torch
 import torch.nn as nn
 
+from fl4health.model_bases.parallel_split_models import ParallelSplitHeadModule, ParallelSplitModel
 from fl4health.model_bases.partial_layer_exchange_model import PartialLayerExchangeModel
 
 
-class FendaJoinMode(Enum):
-    CONCATENATE = "CONCATENATE"
-    SUM = "SUM"
-
-
-class FendaHeadModule(nn.Module, ABC):
-    def __init__(self, mode: FendaJoinMode) -> None:
-        super().__init__()
-        self.mode = mode
-
-    @abstractmethod
-    def local_global_concat(self, local_tensor: torch.Tensor, global_tensor: torch.Tensor) -> torch.Tensor:
-        raise NotImplementedError
-
-    @abstractmethod
-    def head_forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
-        raise NotImplementedError
-
-    def forward(self, local_tensor: torch.Tensor, global_tensor: torch.Tensor) -> torch.Tensor:
-        head_input = (
-            self.local_global_concat(local_tensor, global_tensor)
-            if self.mode == FendaJoinMode.CONCATENATE
-            else torch.add(local_tensor, global_tensor)
+class FendaModel(PartialLayerExchangeModel, ParallelSplitModel):
+    def __init__(self, local_module: nn.Module, global_module: nn.Module, model_head: ParallelSplitHeadModule) -> None:
+        ParallelSplitModel.__init__(
+            self, first_module=local_module, second_module=global_module, model_head=model_head
         )
-        return self.head_forward(head_input)
 
+    def layers_to_exchange(self) -> List[str]:
+        return [layer_name for layer_name in self.state_dict().keys() if layer_name.startswith("second_module.")]
 
-class FendaModel(PartialLayerExchangeModel):
-    def __init__(self, local_module: nn.Module, global_module: nn.Module, model_head: FendaHeadModule) -> None:
-        super().__init__()
-        self.local_module = local_module
-        self.global_module = global_module
-        self.model_head = model_head
 
-    def layers_to_exchange(self) -> List[str]:
-        return [layer_name for layer_name in self.state_dict().keys() if layer_name.startswith("global_module.")]
+class FendaModelWithFeatureState(FendaModel):
```

Comment:
I will leave it up to your judgement because I don't want you to make documentation for the sake of making documentation but I think for at least some of the model bases you added it mat be helpful to add some documentation to help people figure out what it is and what context it should be used in

Reply:
Yeah I think that's a good call. Will add some here.